examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,191 @@
+      (int)data->message->payloadlen, (char *)data->message->payload);
+}
+
+static int ifconfig(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    for (struct netif *iface = netif_list; iface != NULL; iface = iface->next) {
+        printf("%s_%02u: ", iface->name, iface->num);
+#ifdef MODULE_LWIP_IPV6
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+        for (int i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+            if (!ipv6_addr_is_unspecified((ipv6_addr_t *)&iface->ip6_addr[i])) {
+                printf(" inet6 %s\n", ipv6_addr_to_str(addrstr, (ipv6_addr_t *)&iface->ip6_addr[i],
+                                                       sizeof(addrstr)));
+            }
+        }
+#endif
+#ifdef MODULE_LWIP_IPV4
+        char addrstr[IPV4_ADDR_MAX_STR_LEN];
+        printf(" inet %s\n", ipv4_addr_to_str(addrstr, (ipv4_addr_t *)&iface->ip_addr,
+                                              sizeof(addrstr)));
+#endif
+        puts("");
+    }
+    return 0;
+}
+
+static void *_mqtt_sender(void *arg)
+{
+    (void) arg;
+    int count = 1;
+    for (;;) {
+        MQTTMessage message;
+        char payload[30];
+
+        message.qos = 1;
+        message.retained = 0;
+        message.payload = payload;
+        sprintf(payload, "message number %d", count);
+        message.payloadlen = strlen(payload);
+
+        int rc;
+        if ((rc = MQTTPublish(&client, topic, &message)) != 0) {
+            printf("mqtt_example: return code from MQTT publish is %d\n", rc);
+        } else {
+            printf("Message (%s) has been published to topic:%s\n", (char *)message.payload, topic);
+        }
+        if (count >= MAX_MSGS) {
+            /* kill thread */
+            printf("mqtt_example: finishing example, check that %d messages were sent/received\n",
+                     count);
+            return NULL;
+        }
+        count++;
+    }
+    return NULL;
+}
+
+static void usage(void)
+{
+    puts("Usage:");
+    puts("  help: this help");
+    puts("  pub msg: publish 'msg' to some random 'topic'");
+}
+
+static int mqtt_publish_cmd(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+
+    if (argc < 2) {
+        usage();
+        return 1;
+    }
+
+#define STR_LEN     25
+    MQTTMessage msg;
+    msg.qos = QOS1;
+    if (strcmp(argv[0], "pub") == 0) {
+        msg.payload = argv[1];
+        msg.payloadlen = strlen(argv[1]);
+        MQTTPublish(&client, topic, &msg);
+        printf("%s message to topic %s\n", argv[1], topic);
+    }
+    return 0;
+}
+
+int main(void)
+{
+    int ret = -1;
+
+    static const shell_command_t shell_commands[] = {
+        { "pub", "publish MQTT 'msg' to 'topic'", mqtt_publish_cmd},
+        { "ifconfig", "Shows assigned IPv6 addresses", ifconfig },
+        { NULL,  NULL, NULL}
+    };
+
+    char* remote_ip = REMOTE_IP_ADDRESS;
+    int port = 8808;
+    char* clientid = "myid";
```

Comment:
It would be great if those hardcoded values could be configured somehow:
- they could be set via CFLAGS macros
- even better, add new commands in the shell to connect, subscribe, start the background publish task. This option would make the application much more flexible and easy to test.